pkg/roll/roll.go

```diff
@@ -44,10 +45,23 @@ func New(ctx context.Context, pgURL, schema string, lockTimeoutMs int, state *st
 		return nil, fmt.Errorf("unable to set lock_timeout: %w", err)
 	}
 
+	var pgMajorVersion string
+	err = conn.QueryRowContext(ctx, "SELECT split_part(split_part(version(), ' ', 2), '.', 1)").Scan(&pgMajorVersion)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve postgres version: %w", err)
+	}
+
+	var viewCreator viewCreator
+	viewCreator = defaultViewCreator{db: conn, schema: schema}
+	if pgMajorVersion == "14" {
+		viewCreator = pg14ViewCreator{db: conn, schema: schema}
+	}
+
 	return &Roll{
-		pgConn: conn,
-		schema: schema,
-		state:  state,
+		pgConn:      conn,
+		schema:      schema,
+		state:       state,
+		viewCreator: viewCreator,
```

Comment:
not against this but I guess that we will do conditional statements based on the version in many places. Perhaps it would be good to store the version as a field here and read it from the createViews function to use one creation mode or another